### PR TITLE
Feature | Add support for annotations

### LINKED ISSUE #57: Introduce `annotations` to SLO metadata
### Problem to solve / Proposal

Hey, similar to k8s annotation system, introduce annotations to OpenSLO, where we can pass metadata to OpenSLO implementations. We have a use case where we want to have annotations for Cloudwatch or any kind of metrics publisher, that we would like to set on SLO level. By adding this the Spec is more customizable and by default is not adding any new implementation details.

```yaml
---
apiVersion: openslo/v1alpha
kind: SLO
metadata:
  name: checkout-front-end-latency
  displayName: Checkout Front End Latency
  annotations:
    paddle.com/cloudwatch-publisher-enabled: true
    paddle.com/cloudwatch-namespace: SLI
```

### PR DESCRIPTION
**Pull request by geototti21**
Add support for annotations, I used [K8s spec](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set) to avoid reinventing the wheel 😆 

Closes #57

### CODE REVIEW

**Comment by nicolastakashi** [comment]
Congrats on the job @geototti21.
After the PR completion, the Oslo should be updated right?

**Comment by ian-bartholomew** [comment]
@nicolastakashi yah, we woud need to update oslo after this

@geototti21 how do you see the difference between `annotations` and something like `labels`?

**Comment by geototti21** [comment]
>how do you see the difference between annotations and something like labels?

@ian-bartholomew I see that we will be able to use the `annotations` to pass down system/implementation-specific information.

eg
```yaml
---
apiVersion: openslo/v1alpha
kind: SLO
metadata:
  name: checkout-front-end-latency
  displayName: Checkout Front End Latency
  annotations:
    paddle.com/cloudwatch-publisher-enabled: "true"
    paddle.com/cloudwatch-namespace: "SLI"
```

For `labels` pretty much on how you describe it on https://github.com/OpenSLO/OpenSLO/pull/52, something that is designed for the end-users of the SLO when we have a UI. Also, I can see labels usage to create links with other SLOs / SLIs if we end up splitting the two concepts into to two files (like mentioned on https://github.com/OpenSLO/OpenSLO/issues/39)

**Comment by ian-bartholomew** [comment]
@geototti21 that sounds good.  Can you add that language to the docs in the PR?  Thanks!

File: .markdownlint.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "default": true,
     "MD013": {
-        "line_length": 120
+        "line_length": 140
```
Comment (geototti21) [inline .markdownlint.json:4]: This is needed as adding inline comments to a yaml its not very flexible 😄

Comment (weyert) [reply .markdownlint.json:4]: Haha nice

**Review by weyert — APPROVED** [review]

**Review by nobl9-mikec — APPROVED** [review]
love this feature! Great way to define computer friendly metadata.  Any specific examples you've already got in mind?

**Pull request merged**